workers/unity/Assets/Gdk/Android/Utility/DeviceInfo.cs

```diff
@@ -0,0 +1,29 @@
+using UnityEngine;
+
+namespace Improbable.Gdk.Android
+{
+    public class DeviceInfo
+    {
+        private static readonly AndroidJavaObject Build = new AndroidJavaObject("android.os.Build");
+        private static readonly string Fingerprint = GetValue("FINGERPRINT");
+        private static readonly string Model = GetValue("MODEL");
+        private static readonly string Manufacturer = GetValue("MANUFACTURER");
+        private static readonly string Brand = GetValue("BRAND");
+        private static readonly string Device = GetValue("DEVICE");
+        private static readonly string Product = GetValue("PRODUCT");
+
+        public static bool IsEmulator()
+        {
+            return Fingerprint.StartsWith("generic") || Fingerprint.StartsWith("unknown") ||
+                Model.Contains("google_sdk") || Model.Contains("Emulator") ||
+                Model.Contains("Android SDK built for x86") || Manufacturer.Contains("Genymotion") ||
```

Comment:
Will this always work on other operating systems? Might be good to make it more general if possible.

Reply:
Resolved offline. The code checks if the device is an emulator of any type (not just Android Studio) and is an implementation that is widely used in different Mobile frameworks. As it is a reflective java call it works on Windows/Mac/Linux.

Reply:
Would be nice to have a comment enumerating which emulators does this code cover. Later on it would be easier to check if particular emulator is covered or not